hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -0,0 +1,48 @@
+module Hydra.API.ServerSpec where
+
+import Cardano.Prelude hiding (atomically, threadDelay)
+import Control.Monad.Class.MonadAsync (concurrently_)
+import Control.Monad.Class.MonadSTM (TQueue, atomically, newTQueue, readTQueue, tryReadTQueue, writeTQueue)
+import Control.Monad.Class.MonadTimer (threadDelay)
+import qualified Data.Text as Text
+import Hydra.API.Server (APIServerLog, withAPIServer)
+import Hydra.HeadLogic (ClientResponse (..))
+import Hydra.Ledger.Mock (MockTx)
+import Hydra.Logging (Tracer (..))
+import Network.WebSockets (runClient)
+import Network.WebSockets.Connection (receiveData)
+import Test.Hspec
+import Test.Util (failAfter)
+
+spec :: Spec
+spec = describe "API Server" $ do
+  it "sends response to all connected clients" $ do
+    queue <- atomically newTQueue
+    failAfter 5 $
+      withAPIServer @MockTx "127.0.0.1" 54321 showStdoutTracer noop $ \response -> do
+        threadDelay 0.5
+        withAsync (concurrently_ (testClient queue) (testClient queue)) $ \_ -> do
+          threadDelay 0.5
+          response ReadyToCommit
+
+          atomically (replicateM 2 (readTQueue queue)) `shouldReturn` [ReadyToCommit, ReadyToCommit]
+          atomically (tryReadTQueue queue) `shouldReturn` Nothing
+
+showStdoutTracer :: Tracer IO APIServerLog
+showStdoutTracer = Tracer print
+
+noop :: Applicative m => a -> m ()
+noop = const $ pure ()
+
+testClient :: HasCallStack => TQueue IO (ClientResponse MockTx) -> IO ()
+testClient queue =
+  runClient
+    "127.0.0.1"
+    54321
```

Comment:
What about introducing some function to our test utils to get a random available port and avoid possible conflicts between tests? It's likely fine for now since we don't have _too many_ tests, but it may become annoying. 

Some pointers / things we can shamelessly steal:

- https://github.com/input-output-hk/cardano-wallet/blob/ca9cbedbf253afdb4623c74d7d137a68317985ad/lib/core/src/Cardano/Wallet/Network/Ports.hs#L103-L112
- https://github.com/input-output-hk/cardano-wallet/blob/ca9cbedbf253afdb4623c74d7d137a68317985ad/lib/core/src/Cardano/Wallet/Network/Ports.hs#L137-L148

Reply:
Couldn't agree more, that's also what I usually do :)

Reply:
100% agree. What I usually do is making it possible to pass a randomly assigned socket whose port is retrieved.